### PR TITLE
More e3sm build upgrades

### PR DESCRIPTION
**Pull request by jgfouca**
Change list:
1) Make time gathering for cmake and make commands more robust.
2) When saving build provenance, collect and save the times needed to build targets to a "build_times.txt" file in similar approach to how build_environment.txt is saved.

Test suite: scripts_regression_tests, by-hand
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes [CIME Github issue #]

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

**Comment by jgfouca** [comment]
@sarats ^

File: scripts/lib/CIME/provenance.py [hunk 1]
```diff
@@ -71,9 +86,18 @@ def _save_build_provenance_e3sm(case, lid):
...
     env_module.save_all_env_info(env_prov)
 
+    # Save build times
+    build_times = os.path.join(exeroot, "build_times.{}.txt".format(lid))
+    if os.path.exists(build_times):
+        os.remove(build_times)
+    globstr = "{}/*bldlog*{}.gz".format(exeroot, lid)
+    matches = glob.glob(globstr)
+    if matches:
+        _extract_times(matches, build_times)
```
Comment (sarats) [inline scripts/lib/CIME/provenance.py:96]: I presume this means the overall component build times are written at the end of this file, right?

Comment (jgfouca) [reply scripts/lib/CIME/provenance.py:97]: @sarats , remember that, with CMake, the entire build is done at once, not component-by-component (unless --separate-builds is provided to case.build). So we can't code our provenance to expect per-component build times.

Comment (sarats) [reply scripts/lib/CIME/provenance.py:97]: Oh right, I was looking at Az's output that probably used separate builds.

**Review by sarats — APPROVED** [review]

**Comment by worleyph** [comment]
@jgfouca , I will be unable to review this for 2-3 weeks. Best to remove me as a reviewer.

**Pull request merged**